alkira/resource_alkira_probe_http.go

```diff
@@ -0,0 +1,190 @@
+							Required:    true,
+						},
+					},
+				},
+			},
+			"uri": {
+				Description: "The URI to probe.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"headers": {
+				Description: "HTTP headers to include in the request.",
+				Type:        schema.TypeMap,
+				Optional:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			"validators": {
+				Description: "Validators for the HTTP response.",
+				Type:        schema.TypeList,
+				Optional:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"type": {
+							Description: "Type of Validator, can be one of " +
+								"`HTTP_STATUS_CODE` or `HTTP_RESPONSE_BODY`.",
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringInSlice([]string{"HTTP_STATUS_CODE", "HTTP_RESPONSE_BODY"}, false),
+						},
+						"status_code": {
+							Type:          schema.TypeString,
+							Description:   "Response code the response should have.",
+							Optional:      true,
+							ConflictsWith: []string{"regex"},
+						},
+						"regex": {
+							Description:   "Regex the response body should match.",
+							Type:          schema.TypeString,
+							ConflictsWith: []string{"status_code"},
+							Optional:      true,
+						},
+					},
+				},
+			},
+			"failure_threshold": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Description: "The number of consecutive failures required to mark the probe as failed.",
+			},
+			"success_threshold": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Description: "The number of consecutive successes required to mark the probe as successful.",
+			},
+			"period_seconds": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Description: "How often (in seconds) to perform the probe.",
+			},
+			"timeout_seconds": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Description: "Number of seconds after which the probe times out.",
+			},
+		},
+	}
+}
+
+func resourceProbeHTTPCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	client := m.(*alkira.AlkiraClient)
+	api := alkira.NewProbe(client)
+
+	probe, err := generateHTTPProbeRequest(d)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	response, _, err, _ := api.Create(probe)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(response.ID)
+
+	return resourceProbeHTTPRead(ctx, d, m)
+}
+
+func resourceProbeHTTPRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	client := m.(*alkira.AlkiraClient)
+	api := alkira.NewProbe(client)
+
+	probe, _, err := api.GetById(d.Id())
+	if err != nil {
+		return diag.Diagnostics{{
+			Severity: diag.Warning,
+			Summary:  "Failed to Get HTTP Probe",
+			Detail:   err.Error(),
+		}}
+	}
+
+	if probe.Type != "HTTP" {
+		return diag.Errorf("Retrieved probe is not of HTTP type")
```

Comment:
This error will be just confusing... I think we should just say type mis-match...